File: informatykaEuropejczyka/Rozdzial2/prog2_28.py
#obliczanie pola obszaru ograniczonego wykresem funkcji, osia 0x i prostymi p, q - metoda trapezow

from math import *

def F (x):
    return -x**3-x**2+1

def oblicz (p, q, n):
    dl=(q-p)/n
    s=0
    for i in range(n):
        s+=fabs(F(p+i*dl))
    return dl/2*(fabs(F(p))+fabs(F(q))+2*s)

#print(oblicz(-2,5,30))

File: informatykaEuropejczyka/Rozdzial2/prog2_28.cpp
//obliczanie pola obszaru ograniczonego wykresem funkcji, osia 0x i prostymi p, q - metoda trapezow
#include <iostream>
#include <cmath>
using namespace std;

double F (double x)
{
 return -pow(x,3)-pow(x,2)+1;
}

double oblicz (double p, double q, int n)
{
 double dl=(q-p)/n, s=0;
 for (int i=1;i<n;i++) s+=abs(F(p+i*dl));
 return dl/2*(abs(F(p))+abs(F(q))+2*s);
}

int main()
{
 double p, q;
 int n;
 cout<<"podaj przedzial [p, q]: ";
 cin>>p>>q;
 cout<<"podaj liczbe prostokatow n: ";
 cin>>n;
 cout<<"pole = "<<oblicz(p,q,n)<<endl;
 return 0;
}
